apps/game_backend/priv/repo/migrations/20240509225708_add_quest_table.exs

```diff
@@ -0,0 +1,19 @@
+defmodule GameBackend.Repo.Migrations.AddQuestTable do
+  use Ecto.Migration
+
+  def change do
+    create table(:quests) do
+      add :description, :string, default: ""
+      add :type, :string
+      add :config_id, :integer
+      add :target, :integer
+      add :objective,  :map
+      add :reward, :map
```

Comment:
Looking at the config seems it could be 2 fields here `reward_currency_id` and `reward_amount`, why not that?

Reply:
Is there any benefit of having this columns over a map?